src/controller/python/chip/yaml/parser.py

```diff
@@ -130,33 +292,38 @@ def run_action(self, dev_ctrl: ChipDeviceCtrl, endpoint: int, node_id: int):
 class ReadAttributeAction(BaseAction):
     '''Single read attribute action to be executed including validation.'''
 
-    def __init__(self, item: dict, cluster: str, data_model_lookup: DataModelLookup):
+    def __init__(self, item: dict, cluster: str, reference_objects: _ReferenceObjects):
         '''Parse read attribute action from yaml test configuration.
 
         Args:
           'item': Dictionary contains single read attribute action to be parsed.
           'cluster': Name of cluster read attribute action is targeting.
-          'data_model_lookup': Data model lookup to get attribute object.
+          'reference_objects': Contains global common objects such has data model lookup, storage
+            for device responses and top level test configurations variable.
         Raises:
           ParsingError: Raised if there is a benign error, and there is currently no
             action to perform for this read attribute.
           UnexpectedParsingError: Raised if there is an unexpected parsing error.
         '''
         super().__init__(item['label'])
         self._attribute_name = stringcase.pascalcase(item['attribute'])
+        self._constraints = None
         self._cluster = cluster
         self._cluster_object = None
         self._request_object = None
         self._expected_raw_response: dict = field(default_factory=dict)
-        self._expected_response_object: None = None
+        self._expected_response: _ExpectedResponse = None
```

Comment:
Suggest storage the reference_objects as a field altogether as well, rather than storing references to many attibutes thereof. If you add items to the reference_objects/contexts, there is less "trampoline fields" to add.

In fact, the reference objects (execution context) could be part of BaseAction base class constructor